modules/framework/galasa-parent/dev.galasa.framework.api.runs/src/main/java/dev/galasa/framework/api/runs/common/GroupRuns.java

```diff
@@ -40,60 +44,76 @@ public class GroupRuns extends ProtectedRoute {
     protected IFramework framework;
     private IResultArchiveStore rasStore;
     private final GalasaGson gson = new GalasaGson();
-    private final ITimeService timeService ;
+    private final ITimeService timeService;
+    private final Environment env;
 
     public GroupRuns(ResponseBuilder responseBuilder, String path, IFramework framework, ITimeService timeService) throws RBACException {
         super(responseBuilder, path, framework.getRBACService());
         this.framework = framework;
         this.rasStore = framework.getResultArchiveStore();
         this.timeService = timeService;
+        this.env = new SystemEnvironment();
     }
 
     protected List<IRun> getRuns(String groupName) throws InternalServletException {
-         List<IRun> runs = null;
+        List<IRun> runs = null;
         try {
             runs = this.framework.getFrameworkRuns().getAllGroupedRuns(groupName);
         } catch (FrameworkException fe) {
-            ServletError error = new ServletError(GAL5019_UNABLE_TO_RETRIEVE_RUNS, groupName);  
+            ServletError error = new ServletError(GAL5019_UNABLE_TO_RETRIEVE_RUNS, groupName);
             throw new InternalServletException(error, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, fe);
         }
         return runs;
     }
 
-    protected ScheduleStatus serializeRuns(@NotNull List<IRun> runs) {
-       
+    protected ScheduleStatus serializeRuns(@NotNull List<IRun> runs, Environment env) {
+
         ScheduleStatus status = new ScheduleStatus();
         boolean complete = true;
         for (IRun run : runs) {
             ScheduleRunCompleteStatus runstatus = ScheduleRunCompleteStatus.getFromString(run.getStatus());
-            if (runstatus ==null) {
+            if (runstatus == null) {
                 complete = false;
             }
 
-            status.getRuns().add(run.getSerializedRun());
+            String runId = run.getRasRunId();
+
+            String baseWebUiUrl = env.getenv(EnvironmentVariables.GALASA_EXTERNAL_WEBUI_URL);
+            String webUiUrl = baseWebUiUrl + "/test-runs/" + runId;
+
+            String baseServletUrl = env.getenv(EnvironmentVariables.GALASA_EXTERNAL_API_URL);
+            String restApiUrl = baseServletUrl + "/ras/runs/" + runId;
+
+            Run serializedRun = run.getSerializedRun();
+            serializedRun.setWebUiUrl(webUiUrl);
+            serializedRun.setRestApiUrl(restApiUrl);
+
+            status.getRuns().add(serializedRun);
```

Comment:
Lets put this into a method with the code in my next comment to avoid code repetition